rules-java/tests/src/test/java/org/jboss/windup/rules/java/JavaClassSourceMatchTest.java

```diff
@@ -0,0 +1,229 @@
+
+            GraphService<JavaTypeReferenceModel> typeRefService = new GraphService<>(context,
+                        JavaTypeReferenceModel.class);
+            Iterable<JavaTypeReferenceModel> typeReferences = typeRefService.findAll();
+            Assert.assertTrue(typeReferences.iterator().hasNext());
+
+            Assert.assertEquals(1, provider.getFirstRuleMatchCount());
+            Assert.assertEquals(1, provider.getSecondRuleMatchCount());
+            Assert.assertEquals(2, provider.getThirdRuleMatchCount());
+        }
+    }
+
+    /**
+     * Testing that .from() and .as() sets the right variable
+     */
+    @Test
+    public void testJavaClassInputOutputVariables()
+    {
+        JavaClass as = (JavaClass) JavaClass.from("input").references("abc").as("output");
+        Assert.assertEquals("input", as.getInputVariablesName());
+        Assert.assertEquals("output", as.getOutputVariablesName());
+    }
+
+    private Path getDefaultPath()
+    {
+        return FileUtils.getTempDirectory().toPath().resolve("Windup")
+                    .resolve("windupgraph_javaclasstest_" + RandomStringUtils.randomAlphanumeric(6));
+    }
+
+    @Singleton
+    public static class JavaClassTestRuleProvider extends WindupRuleProvider
+    {
+        private static Logger log = Logger.getLogger(RuleSubset.class.getName());
+
+        private int thirdRuleMatchCount = 0;
+        private int firstRuleMatchCount = 0;
+        private int secondRuleMatchCount = 0;
+
+        // @formatter:off
+        @Override
+        public Configuration getConfiguration(GraphContext context)
+        {
+            return ConfigurationBuilder.begin()
+            .addRule().when(JavaClass.references("org.jboss.windup.graph.model.resource.FileModel.setFilePath{*}").matchesSource("{*}/JavaHintsClassificationsTest.java{*}").inType("{*}").at(TypeReferenceLocation.METHOD_CALL))
+            .perform( Iteration.over().perform(new AbstractIterationOperation<JavaTypeReferenceModel>()
+                        {
+                @Override
+                public void perform(GraphRewrite event, EvaluationContext context, JavaTypeReferenceModel payload)
+                {
+                    firstRuleMatchCount++;
+                }
+            }).endIteration())
+            .addRule()
+            .when(JavaClass.references("org.jboss.windup.rules.java.JavaClassTestFile1.testJavaClassCondition()").matchesSource("{*}{line}{*}").inType("{*}").at(TypeReferenceLocation.METHOD))
+            .perform( Iteration.over().perform(new AbstractIterationOperation<JavaTypeReferenceModel>()
+                        {
+                @Override
+                public void perform(GraphRewrite event, EvaluationContext context, JavaTypeReferenceModel payload)
+                {
+                    secondRuleMatchCount++;
+                }
+            }).endIteration())
+            .where("line").matches("testJavaClassCondition\\(\\) throws IOException, InstantiationException")
```

Comment:
You have a "\n" character at the beginning and that is throwing off the regex match. Seems a little brittle doesn't it? :)